docs/manifest.md

```diff
@@ -0,0 +1,104 @@
+# Manifest
+
+A Fondant manifest describes the contents of a dataset and is used as a reference to the dataset 
+to be passed between Fondant components.
+
+Creating the manifest is handled by Fondant, as each component will automatically generate the 
+manifest that represents its output dataset. While you should not have to create the manifest 
+yourself, it is still useful to understand how it works.
+
+## Contents
+
+A manifest consists of the following sections
+
+```json
+{
+  "metadata": {
+    ...
+  },
+  "index": {
+    ...
+  },
+  "subsets": {
+    ...
+  }
+}
+```
+
+### Metadata
+
+The metadata section tracks metadata about the dataset. Currently, it tracks the location where 
+the data is stored and references to the pipeline and component that were used to create it.
+
+```json
+{
+  "metadata": {
+    "base_path": "gs://bucket",
+    "run_id": "12345",
+    "component_id": "67890"
+  }
+}
+```
+
+This metadata is currently only used to build the location where the data should be stored, but 
+can in the future be extended and used for more advanced use cases such as data lineage and 
+pipeline caching.
+
+### Index
+
+The index represents which data points are part of the dataset. The index is tracked separately 
+from the data, so data points can be filtered out of a dataset without actually having to touch 
+the underlying data. The index is stored in a fixed format, which is why the manifest only 
+tracks the location where it is saved:
+
+```json
+{
+  "index": {
+    "location": "/index"
+  }
+}
+```
+
+### Subsets
+
+Each subset is a different view on the data contained in the dataset. Different subsets could 
```

Comment:
Wouldn't say it's a different view, it's just a part of the dataset

Reply:
Reworded into "different features".